### PR TITLE
732 upgrade issue fixes

### PR DESCRIPTION
**Pull request by sonyDeswal**
1) ContentStatus History table is not getting Comments field size, resized to 500...
2) sys_lasttransition field is not showing up in Content Types Global Configuration Fields.
 3) Don't overwrite rx_resources\stylesheets
4) Only copy FastForward ContentType if already present..

### CODE REVIEW

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/install.xml [hunk 1]
```diff
@@ -547,7 +553,19 @@
         <exclude name="ObjectStore/**"/>
         <exclude name="jetty/base/logs"/>
         <exclude name="rxconfig/Installer/*.bat"/>
-        <exclude name="ObjectStore/psx_cerff*.xml"/>
+        <exclude name="ObjectStore/psx_cerffAutoIndex.xml"/>
```
Comment (natechadwick) [inline modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/install.xml:556]: 557-568 We don't want to outright exclude these.  We need to overwrite them if they are present on upgrade, and not install them if they are not present.

File: modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcStatementFactory.java [hunk 2]
```diff
@@ -1567,6 +1571,49 @@ private static PSJdbcSqlStatement getAddComponentStatement(String tableName,
       return new PSJdbcSqlStatement(buf.toString());
    }
 
+   /**
+    * Generates an alter column statement to alter column defination as defined.
```
Comment (natechadwick) [inline modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcStatementFactory.java:1575]: Please turn on warnings / spell check on in IDE

File: modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcTableComponent.java [hunk 3]
```diff
@@ -259,10 +259,7 @@ public int hashCode()
     */
    public boolean canAlter()
    {
-      return (getAction() == ACTION_CREATE ||
-         getAction() == ACTION_NONE) ||
-         getAction() == ACTION_DELETE
-         ;
+      return true;
```
Comment (natechadwick) [inline modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcTableComponent.java:262]: Is there no scenario where we can't alter the table?

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/data/cmsTableDef.xml [hunk 4]
```diff
@@ -361,7 +361,7 @@
                 <size>500</size>
                 <allowsnull>yes</allowsnull>
             </column>
-            <column name="TRANSITIONCOMMENT" >
+            <column action="r" name="TRANSITIONCOMMENT" >
```
Comment (natechadwick) [inline modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/data/cmsTableDef.xml:364]: What does "r" do?  Can you add a comment in the XML file here.

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/install.xml [hunk 5]
```diff
@@ -537,6 +537,12 @@
         <exclude name="jetty/base/resources/log4j*.xml"/>
         <exclude name="rxconfig/Installer/rxrepository.properties"/>
         <exclude name="rx_resources/vm/custom_head.vm"/>
+        <exclude name="rx_resources/stylesheets/rx_Templates.xsl"/>
```
Comment (natechadwick) [inline modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/install.xml:540]: Shouldn't this be rx_resources/*?

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/installRepository.xml [hunk 6]
```diff
@@ -329,6 +328,19 @@
                        failonerror="true" silenceerrors="false"
         />
 
+        <PSExecSQLStmt printExceptionStackTrace="true" qualifyTableNames="RXS_CT_GENERIC"
+                       qualifyViewNames=""
+                       sqlSqlServer="sp_rename 'RXS_CT_GENERIC.USAGE', 'PUSAGE', 'COLUMN';"
+                       sql="ALTER TABLE RXS_CT_GENERIC rename column USAGE to PUSAGE;"
+                       failonerror="false" silenceerrors="true"
+        />
+<!-- There is an issue altering this table column to not null, thus taking this hack -->
```
Comment (natechadwick) [inline modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/installRepository.xml:337]: What happens to the existing data in this table on upgrade?

Comment (natechadwick) [reply modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/installRepository.xml:337]: nm I see checking if it is empty.  Won't this still be an issue if the table is not empty?

**Review by natechadwick — COMMENTED** [review]
Left some questions / feedback inline.

File: modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcTableComponent.java (hunk 3 above)
Comment (sonyDeswal) [reply modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcTableComponent.java:262]: Returning true in base class as extended classes are taking care of correct return.

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/data/cmsTableDef.xml (hunk 4 above)
Comment (sonyDeswal) [reply modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/data/cmsTableDef.xml:364]: 'r' means Replace

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/installRepository.xml (hunk 6 above)
Comment (sonyDeswal) [reply modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/installRepository.xml:337]: I removed this and used a different approach to handle this case.

File: modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/data/cmsTableDef.xml (hunk 4 above)
Comment (sonyDeswal) [reply modules/perc-distribution-tree/src/main/resources/distribution/rxconfig/Installer/data/cmsTableDef.xml:364]: Added comment

File: modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcStatementFactory.java (hunk 2 above)
Comment (sonyDeswal) [reply modules/TableFactory/src/main/java/com/percussion/tablefactory/PSJdbcStatementFactory.java:1575]: is there.. but not taking advantage :)

**Comment by sonyDeswal** [comment]
divided this task into multiple small tasks.. thus don't want to merge this pull request

**Pull request closed without merging**